api/api.go

```diff
@@ -101,3 +126,54 @@ func OrbQuery(ctx context.Context, logger *logger.Logger, configPath string) (*C
 			}
 		}`)
 }
+
+func publishOrbQuery(ctx context.Context, logger *logger.Logger, configPath string,
+	orbVersion string, orbID string, response interface{}, query string) error {
+	config, err := loadYaml(configPath)
+	if err != nil {
+		return err
+	}
+
+	request := client.NewAuthorizedRequest(viper.GetString("token"), query)
+	request.Var("config", config)
+	request.Var("orbId", orbID)
+	request.Var("version", orbVersion)
+
+	graphQLclient := client.NewClient(viper.GetString("endpoint"), logger)
+
+	err = graphQLclient.Run(ctx, request, response)
+
+	if err != nil {
+		return errors.Wrap(err, "Unable to validate config")
+	}
+
+	return nil
+}
+
+// OrbPublish publishes a new version of an orb
+func OrbPublish(ctx context.Context, logger *logger.Logger,
+	configPath string, orbVersion string, orbID string) (*PublishOrbResponse, error) {
+	var response struct {
+		PublishOrb struct {
+			PublishOrbResponse
+		}
+	}
+
+	return &response.PublishOrb.PublishOrbResponse, publishOrbQuery(ctx, logger,
```

Comment:
I would delete the `publishOrbQuery` function and just put the body of it here in `OrbPublish`. The reason for the split in the config validation code-path is because I wanted to re-use a large chunk of code that was copy+pasted between `orb validate` and `config validate`.